File: 2780 minimum-index-of-a-valid-split/solution.cs
public class Solution {
    public int MinimumIndex(IList<int> nums) {
        int x = 0,c=0;
        foreach(int i in nums) {
            if (c == 0) {
                x = i;
                c = 1;
            } else if (i == x) {
                c++;
            } else {
                c--;
            }
        }
        c=0;
        foreach(int i in nums) {
            if(x==i) c++;
        }
        if (c <= nums.Count()/2) {
            return -1;
        }
        int l = 0;
        for(int i=0;i<nums.Count()-1;i++) {
            if (nums[i] == x) {
                l++;
            }
            int r = c-l;
            if(l > (i+1)/2 && r > (nums.Count()-(i+1))/2) {
                return i;
            }
        }
        return -1;
    }
}

File: 2780 minimum-index-of-a-valid-split/solution.go
func minimumIndex(nums []int) int {
    cd,c:=0,0
    for _,v:=range nums {
        if c==0 {
            cd=v
            c=1
        } else if cd==v {
            c++ 
        } else {
            c--
        }
    }
    c=0
    for _,v:=range nums {
        if cd == v {
            c++
        }
    }
    if c <= len(nums)/2 {
        return -1
    }
    l:=0
    for i:=0;i<len(nums)-1;i++ {
        if nums[i] == cd {
            l++
        }
        r:=c-l
        if l > (i+1)/2 && r > (len(nums)-(i+1))/2 {
            return i
        }
    }
    return -1
}
